clang/include/clang/Basic/DiagnosticDriverKinds.td

```diff
@@ -354,6 +354,8 @@ def warn_drv_sycl_target_missing : Warning<
   InGroup<SyclTarget>;
 def err_drv_no_rdc_sycl_target_missing : Error<
   "linked binaries do not contain expected '%0' target; found targets: '%1', this is not supported with '-fno-sycl-rdc'">;
+def err_drv_fsycl_wrong_optimization_options : Error<
+  "-fsycl-optimize-framework option can be used only in conjunction with -O0 option">;
```

Comment:
Maybe `err_drv_fsycl_framework_requires_option` and make the option added as a parameter (so you can use the proper option name difference between Linux and MSVC style optimization disabling)

Reply:
Made the debug option a parameter.